src/libraries/System.Private.CoreLib/src/System/Range.cs

```diff
@@ -47,12 +52,17 @@ public Range(Index start, Index end)
         /// <summary>Returns the hash code for this instance.</summary>
         public override int GetHashCode()
         {
+#if SYSTEM_PRIVATE_CORELIB || NETCOREAPP
```

Comment:
In some other places where we've had these one-off experimental downlevel packages, we've instead chosen to just duplicate/specialize the source, with dedicated files for that package, rather than complicating shared code with lots of ifdefs that we'll likely end up removing in the near future. Might want to consider that with this, too.

Reply:
I've simplified the ifdefs in this file, so it looks much better now. I can split the `GetHashCode` and `ToString` out into specialized files if you still think it is worth it.